plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -156,7 +159,13 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []api.Volume, namespace
 			}
 			pvc, err := c.pvcInfo.GetPersistentVolumeClaimInfo(namespace, pvcName)
 			if err != nil {
-				return err
+				// if the PVC is not found, log the error and count the PV towards the PV limit
+				// generate a random volume ID since its required for de-dup
+				glog.Error(err)
```

Comment:
Always util.HandleError

On Mon, Jun 13, 2016 at 10:58 AM, Solly Ross <notifications@github.com>
wrote:

> In plugin/pkg/scheduler/algorithm/predicates/predicates.go
> <https://github.com/kubernetes/kubernetes/pull/27227#discussion_r66806619>
> :
>
> > @@ -156,7 +159,13 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []api.Volume, namespace
> >  			}
> >  			pvc, err := c.pvcInfo.GetPersistentVolumeClaimInfo(namespace, pvcName)
> >  			if err != nil {
> > -				return err
> > +				// if the PVC is not found, log the error and count the PV towards the PV limit
> > +				// generate a random volume ID since its required for de-dup
> > +				glog.Error(err)
>
> it might not be a bad idea here to make this an Errorf and say something
> like
> glog.Errorf("Unable to look up PVC info for %s/%s, assuming PVC matches
> predicate when counting limits: %v", namespace, pvcName, err), that way
> we can see what the context of the error is from the logs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27227/files/0bc8858a4fa5a2b64df7995ff3709bbbd3f8127a#r66806619>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_pyFvmxtzUtFN3qg8UwHL9qMuTF0Eks5qLW_-gaJpZM4Izcbm>
> .
>